woocommerce/woocommerce-android
# Free Trial: Introduce More menu header badge

Summary
==========
Fix issue #8842 by introducing a badge to the More Menu view that follows the current Woo Express plan used. If there's no site plan, the badge doesn't show up. It also refactor some parts of the Site Plan to centralize the plan name acquisition logic between the Upgrades and More Menu views.

Screenshots
==========
| Light mode  | Dark mode |
| ------------- | ------------- |
| ![Screenshot_20230420_024811](https://user-images.githubusercontent.com/5920403/233269774-a088baee-6086-4341-923d-485e542b265c.png) | ![Screenshot_20230420_024821](https://user-images.githubusercontent.com/5920403/233269780-8cbd77f6-97a9-4b4f-87cf-bff03592fb01.png) |

How to Test
==========
1. Open the app with three different site types: Woo Express Free Trial, Woo Express Subscribed, Atomic Woo store
2. For the Woo Express Free Trial store, verify that the More Menu badge is `Free Trial`
3. For the Woo Express Subscribed, verify that the More menu badge is the plan name. E.g. `Performance`
4. For the Atomic Woo store, verify that the More menu badge is not visible

Update release notes:
